vllm/model_executor/layers/linear.py

```diff
@@ -363,15 +375,22 @@ def __init__(
         input_size = self.hidden_size
         output_size = (self.num_heads +
                        2 * self.num_kv_heads) * tp_size * self.head_size
+
         super().__init__(input_size, output_size, bias, False, skip_bias_add,
-                         params_dtype, linear_method)
+                         params_dtype, linear_method, [
+                             self.num_heads * tp_size * self.head_size,
+                             self.num_kv_heads * tp_size * self.head_size,
+                             self.num_kv_heads * tp_size * self.head_size
+                         ])
```

Comment:
I would rather have done `[self.num_heads, self.num_kv_heads, self.num_kv_heads] * (tp_size * self.head_size)` but of course that just replicates rather than multiplying the interior.  The style seems to be to fully expand this kind of equation rather than, say, turn them into numpy tensors and do math on them, so I just left it expanded as you see.